precompile/allowlist/allowlist.go

```diff
@@ -164,9 +170,14 @@ func CreateAllowListPrecompile(precompileAddr common.Address) contract.StatefulP
 
 func CreateAllowListFunctions(precompileAddr common.Address) []*contract.StatefulPrecompileFunction {
 	setAdmin := contract.NewStatefulPrecompileFunction(setAdminSignature, createAllowListRoleSetter(precompileAddr, AdminRole))
+	setManager := contract.NewStatefulPrecompileFunctionWithActivator(setManagerSignature, createAllowListRoleSetter(precompileAddr, ManagerRole), isManagerRoleActivated)
 	setEnabled := contract.NewStatefulPrecompileFunction(setEnabledSignature, createAllowListRoleSetter(precompileAddr, EnabledRole))
 	setNone := contract.NewStatefulPrecompileFunction(setNoneSignature, createAllowListRoleSetter(precompileAddr, NoRole))
 	read := contract.NewStatefulPrecompileFunction(readAllowListSignature, createReadAllowList(precompileAddr))
 
-	return []*contract.StatefulPrecompileFunction{setAdmin, setEnabled, setNone, read}
+	return []*contract.StatefulPrecompileFunction{setAdmin, setManager, setEnabled, setNone, read}
+}
+
+func isManagerRoleActivated(evm contract.AccessibleState) (bool, error) {
+	return evm.GetChainConfig().IsDUpgrade(evm.GetBlockContext().Timestamp()), nil
```

Comment:
does this need an error in the return type?

Reply:
nope, changed.